types/pouchdb-core/index.d.ts

```diff
@@ -555,9 +550,9 @@ declare namespace PouchDB {
          *
          * @see {@link https://pouchdb.com/2014/06/17/12-pro-tips-for-better-code-with-pouchdb.html|PouchDB Pro Tips}
          */
-        post(doc: Core.PostDocument<Content>,
-            options: Core.PostOptions | null,
-            callback: Core.Callback<Core.Error, Core.Response>): void;
+        post<Model>(doc: Core.PostDocument<Content & Model>,
+                    options: Core.Options | null,
+                    callback: Core.Callback<Core.Response>): void;
```

Comment:
I removed the error type parameter from the `Core.Callback`. It was inconsistent where it was `any` or `Core.Error` which at the time was an empty interface, and those were the only two possibilities returned in the current usage of `Core.Callback`. I added all of the most common error fields across the PouchDB API documentation to the `Core.Error` interface and made it the only Error returned by the `Core.Callback` rather than making it a generic option. That type parameter could be added back in if someone wished to add more specific Error objects, but for the time being it was mostly an unused option and leak for explicit `any` objects. I thought the "grab-bag `Error` object" was a reasonable approach in this case, largely because that's about the extent of documented behavior of errors in PouchDB.